src/MovingAverage.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Moving Average Filter (linear FIR filter with equal weights)
+ */
+#include "iarc7_sensors/MovingAverage.hpp"
+
+#include <ros/ros.h>
+
+namespace iarc7_sensors {
+
+MovingAverage::MovingAverage(const ros::Duration& time) : filter_time_constant_(time) {}
+
+void MovingAverage::updateFilter(double value, double derivative, const ros::Time& time) {
+    if (!value_buffer_.empty() && time <= value_buffer_.back().stamp) {
+        ROS_ERROR("Tried to add point to filter with timestamp older than newest point in filter");
+        return;
+    }
+
+    value_buffer_.emplace_back(value, time);
+
+    while (value_buffer_.size() > 1) {
+        ros::Duration time_since_second_reading =
+            value_buffer_.back().stamp - value_buffer_[1].stamp;
```

Comment:
When the left edge of the filter is between samples, we want to have one sample that's actually outside of the filter range so we can interpolate between those two samples.  At least that's how I did it.  That condition requires throwing out data in the way that I have here.